src/osm_flex/download.py

```diff
@@ -36,27 +37,60 @@ def _create_gf_download_url(iso3, file_format):
 
     Returns
     -------
-    str : Geofabrik download-string for the requested country.
+    url: str
+        Geofabrik download URL for the requested country.
 
     See also
     --------
     DICT_GEOFABRIK for exceptions / special regions.
     """
+    # Retrieve Geofabrik definitions
     try:
-        if file_format == 'shp':
-            return f'{GEOFABRIK_URL}{DICT_GEOFABRIK[iso3][0]}/{DICT_GEOFABRIK[iso3][1]}-latest-free.shp.zip'
-        if file_format == 'pbf':
-            return f'{GEOFABRIK_URL}{DICT_GEOFABRIK[iso3][0]}/{DICT_GEOFABRIK[iso3][1]}-latest.osm.pbf'
-    except KeyError:
-        if iso3=='RUS':
-            raise KeyError("""Russia comes in two files. Please specify either
-                         'RUS-A for the Asian or RUS-E for the European part.""")
-        raise KeyError("""The provided iso3 seems not to be available on
-                           Geofabrik.de. You can clip it from the planet
-                           file or an adequate regional file, instead. See
-                           the methods in the clip module for this.""")
-
-    return LOGGER.error('invalid file format. Please choose one of [shp, pbf]')
+        continent, country = DICT_GEOFABRIK[iso3]
+
+    except KeyError as err:
+        if iso3 == "RUS":
+            raise KeyError(
+                "Russia comes in two files. Please specify either RUS-A for the Asian "
+                "or RUS-E for the European part."
+            ) from err
+        raise KeyError(
+            "The provided iso3 seems not to be available on Geofabrik.de. You can clip "
+            "it from the planet file or an adequate regional file, instead. See the "
+            "methods in the clip module for this."
+        ) from err
+
+    # Set file extension
+    if file_format == "shp":
+        ext = "-free.shp.zip"
+    elif file_format == "pbf":
+        ext = ".osm.pbf"
+    else:
+        raise NotImplementedError(
+            f"Invalid file format '{file_format}'. Please choose one of [shp, pbf]"
+        )
+
+    # Join to URL
+    return urljoin(GEOFABRIK_URL, f"{continent}/{country}-latest{ext}")
+
+def _download_file(download_url: str, filepath: Path, overwrite: bool = True):
+    """Download a file located at an URL to a local file path
+
+    Parameters
+    ----------
+    download_url : str
+        URL of the file to download
+    filepath : str or Path
+        Local file path to store the file
+    overwrite : bool, optional
+        Overwrite existing files. If ``False``, the download will be skipped for
+        existing files. Defaults to ``True``.
+    """
+    if not Path(filepath).is_file() or overwrite:
+        LOGGER.info(f"Download file: {filepath}")
+        urllib.request.urlretrieve(download_url, filepath)
+    else:
+        LOGGER.info(f"Skip existing file: {filepath}")
```

Comment:
This is precisely what we should *not* do, see https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library.

We could do the following to add a shortcut for users who want logging:
```python
# osm_flex.__init__.py

import logging

# Default to no logging
logging.getLogger("osm_flex").addHandler(logging.NullHandler())

def enable_logs(level=logging.INFO, **kwargs):
    """Overwrite default and set basig logging config"""
    logging.basicConfig(level=level, force=True, **kwargs)


# my_code.py

import osm_flex

osm_flex.enable_logs()
```

Reply:
ah nice, I like it! but why is the default no logging?

Reply:
The idea is that if you use osm-flex as dependency of a downstream library, you don't want it to set a certain logging configuration that messes up what logging you want to set up. From [here](https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library):

> Note: It is strongly advised that you do not add any handlers other than [NullHandler](https://docs.python.org/3/library/logging.handlers.html#logging.NullHandler) to your library’s loggers. This is because the configuration of handlers is the prerogative of the application developer who uses your library. The application developer knows their target audience and what handlers are most appropriate for their application: if you add handlers ‘under the hood’, you might well interfere with their ability to carry out unit tests and deliver logs which suit their requirements.